src/pages/søknad/steg/bo-og-opphold-i-norge/Bo-og-opphold-i-norge.tsx

```diff
@@ -55,27 +54,42 @@ const schema = yup.object<FormData>({
                 ])
                 .required(),
         }),
-    ektefellePartnerSamboer: yup
-        .mixed<EPSFormData>()
-        .nullable()
-        .defined()
-        .when('delerBoligMed', {
-            is: DelerBoligMed.EKTEMAKE_SAMBOER,
-            then: yup
-                .mixed<EPSFormData>()
-                .required()
-                .test('isValidEktefelleData', 'Ugyldig informasjon om ektefelle', (value) => {
-                    const eps = toEktefellePartnerSamboer(value);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore: Siden EPS er Nullable, med verdier som er nullable, er det litt vanskelig å få riktig feilmelding på riktige inputs.
+    //Yup sin otherwise i when() fanger ikke typingen godt nok, men valideringen i seg selv funker.
+    ektefellePartnerSamboer: yup.object<EPSFormData>().when('delerBoligMed', {
+        is: DelerBoligMed.EKTEMAKE_SAMBOER,
+        then: yup
+            .object<EPSFormData>({
+                fnr: yup
+                    .string()
+                    .required()
+                    .length(11)
+                    .typeError('Feltet må fylles ut')
+                    .test({
+                        name: 'fnr',
+                        message: 'Ugyldig fødselsnummer',
+                        test: function (value) {
+                            if (!value) return false;
 
-                    if (!eps) return false;
-
-                    return eps.fnr.length === 11 && fnrValidator.fnr(eps.fnr).status === 'valid';
-                })
-                .test('UførhetFyltInn', 'Må fylles inn', (value) => {
-                    if (!value) return false;
-                    return value.erUførFlyktning !== null;
-                }),
-        }),
+                            return fnrValidator.fnr(value).status === 'valid';
+                        },
+                    }),
+                erUførFlyktning: yup
+                    .boolean()
+                    .required()
+                    .typeError('Feltet må fylles ut')
+                    .test({
+                        name: 'fnr',
```

Comment:
Skal det være `erUførFlyktning` her?

Og, funker det ikke med `yup.boolean().defined().nullable()` i dette tilfellet?

Reply:
erUførFlyktning ja. 

`yup.boolean().defined().nullable()` for wut exactly?

Reply:
I stedet for å bruke `.test`-greia.

Reply:
ser ut som at vi ikke trenger dem begge😅